aiida/tools/dbimporters/plugins/materialsproject.py

```diff
@@ -0,0 +1,198 @@
+
+        :return: list of strings
+        """
+        return self._supported_keywords
+
+    def query(self, **kwargs):
+        """
+        Query the database with a given dictionary of query parameters for a given properties
+
+        :param query: a dictionary with the query parameters
+        :param properties: the properties to query
+        """
+        try:
+            query = kwargs['query']
+        except AttributeError:
+            raise AttributeError('Make sure the supplied dictionary has `query` as a key. This '
+                                 'should contain a dictionary with the right query needed.')
+        try:
+            properties = kwargs['properties']
+        except AttributeError:
+            raise AttributeError('Make sure the supplied dictionary has `properties` as a key.')
+
+        if not isinstance(query, dict):
+            raise TypeError('The query argument should be a dictionary')
+
+        if properties is None:
+            properties = self._properties
+
+        if properties != 'structure':
+            raise ValueError('Unsupported properties: {}'.format(properties))
+
+        results = []
+        properties_list = ['material_id', 'cif']
+        for entry in self.find(query, properties_list):
+            results.append(entry)
+        search_results = MaterialsProjectSearchResults(results, return_class=MaterialsProjectCifEntry)
+
+        return search_results
+
+    def find(self, query, properties):
+        """
+        Query the database with a given dictionary of query parameters
+
+        :param query: a dictionary with the query parameters
+        """
+        for entry in self._mpr.query(criteria=query, properties=properties):
+            yield entry
+
+
+class MaterialsProjectCifEntry(CifEntry, DbEntry):  # pylint: disable=abstract-method
+    """
+    A Materials Project entry class which extends the DbEntry class with a CifEntry class.
+    """
+
+    _license = 'Materials Project'
+
+    def __init__(self, url, **kwargs):
+        """
+        The DbSearchResults base class instantiates a new DbEntry by explicitly passing the url
+        of the entry as an argument. In this case it is the same as the 'uri' value that is
+        already contained in the source dictionary so we just copy it
+        """
+        cif = kwargs.pop('cif', None)
+        kwargs['uri'] = url
+        super(MaterialsProjectCifEntry, self).__init__(**kwargs)
+
+        if cif is not None:
+            self.cif = cif
+
+
+class MaterialsProjectSearchResults(DbSearchResults):  # pylint: disable=abstract-method
+    """
+    A collection of MaterialsProjectEntry query result entries.
+    """
+
+    _db_name = 'Materials Project'
+    _db_uri = 'https://materialsproject.org'
+    _material_base_url = 'https://materialsproject.org/materials/'
+    _version = str(datetime.datetime.now())
+    _return_class = MaterialsProjectCifEntry
+
+    def __init__(self, results, return_class=None):
```

Comment:
Method ``_init_`` can be removed altogether, as ``return_class`` will always be ``MaterialsProjectCifEntry``

Reply:
See comment above.